### DISCUSSION TITLE
GenPollText: Trigger poll on mouse callback

### DISCUSSION THREAD

**Discussion by myrkvi** [qtile/qtile#3420] [Q&A] [answered]
Hi! Is there any way to cause a GenPollText widget to trigger a poll when a mouse callback occurs?

**Comment by elParaguayo** [comment, marked as answer]
Yes!

Firstly, if you have more than one `GenPollText` widget, you should add a `name` parameter and give them unique names.

Then, you can do this:
``` python
def poll_widget(widget):
    widget.update(widget.poll())

# Your widget mouse_callback would then be
{"Button1": lazy.widget["genpolltext"].function(poll_widget)}
```
